File: console/src/services/console/providerController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utils/request';

/** 此处后端没有提供注释 GET /api/v1/providers */
export async function getProviders(options?: { [key: string]: any }) {
  return request<API.ProviderResponse[]>('/api/v1/providers', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/v1/providers */
export async function createProvider(
  body: API.CreateProviderRequest,
  options?: { [key: string]: any },
) {
  return request<API.ProviderResponse>('/api/v1/providers', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /api/v1/providers */
export async function batchDeleteProvider(
  body: API.BatchDeleteRequest,
  options?: { [key: string]: any },
) {
  return request<any>('/api/v1/providers', {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/providers/${param0} */
export async function getProvider(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getProviderParams,
  options?: { [key: string]: any },
) {
  const { providerId: param0, ...queryParams } = params;
  return request<API.ProviderResponse>(`/api/v1/providers/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/providers/${param0}/models */
export async function getProviderModels(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getProviderModelsParams,
  options?: { [key: string]: any },
) {
  const { providerId: param0, ...queryParams } = params;
  return request<API.ModelResponse[]>(`/api/v1/providers/${param0}/models`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/providers/tree */
export async function getProviderTree(options?: { [key: string]: any }) {
  return request<API.TreeNode[]>('/api/v1/providers/tree', {
    method: 'GET',
    ...(options || {}),
  });
}

File: console-api/src/main/java/io/github/nexllm/console/controller/ProviderController.java
package io.github.nexllm.console.controller;

import io.github.nexllm.console.model.request.CreateProviderRequest;
import io.github.nexllm.console.model.response.ModelResponse;
import io.github.nexllm.console.model.response.ProviderResponse;
import io.github.nexllm.console.model.response.TreeNode;
import io.github.nexllm.console.service.ModelService;
import io.github.nexllm.console.service.ProviderService;
import io.github.nexllm.console.model.request.BatchDeleteRequest;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.validation.Valid;
import java.util.List;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("api/v1/providers")
@RequiredArgsConstructor
public class ProviderController {

    private final ProviderService providerService;
    private final ModelService modelService;

    @PostMapping
    public ProviderResponse createProvider(@RequestBody @Valid CreateProviderRequest createProviderRequest) {
        return providerService.createProvider(createProviderRequest);
    }

    @GetMapping
    public List<ProviderResponse> getProviders() {
        return providerService.getCurrentTenantProviders();
    }

    @GetMapping("tree")
    public List<TreeNode> getProviderTree() {
        return providerService.getProviderTree();
    }

    @GetMapping("{providerId}")
    public ProviderResponse getProvider(@PathVariable("providerId") UUID providerId) {
        return providerService.getProvider(providerId);
    }

    @Operation(operationId = "batchDeleteProvider")
    @DeleteMapping
    @PreAuthorize("@providerService.hasPermissions(#batchDeleteRequest.ids())")
    public void batchDelete(@RequestBody @Valid BatchDeleteRequest batchDeleteRequest) {
        providerService.batchDelete(batchDeleteRequest.ids());
    }

    @GetMapping("{providerId}/models")
    public List<ModelResponse> getProviderModels(@PathVariable("providerId") UUID providerId) {
        return modelService.getProviderModels(providerId);
    }
}
